link.go

```diff
@@ -638,6 +646,19 @@ func (l *link) muxHandleFrame(fr frameBody) error {
 	return nil
 }
 
+// Check checks the link state, returning an error if the link is closed (ErrLinkClosed) or if
+// it is in a detached state (ErrLinkDetached)
+func (l *link) Check() error {
+	select {
+	case <-l.detached:
+		return ErrLinkDetached
```

Comment:
any case where both channel have a value?
if both have an error in it, the returned value will be undefined I think? (either or...)
but, probably does not matter since in both case, the link is not good.

Reply:
I had a similar conversation with @jhendrixMSFT - this is probably something we should discuss more (I think this PR is fine, but it'd be good to think about how we represent it).

For instance, if you received an error and then called 'Check' you'd get back detached. Reacting to that by recreating a link feels right (ie: detached is a retryable error at the link level). 

However, if you were detached and then also called Close() - that's no longer retryable. You should consider that someone is trying to shut down. So I'd like Close() to take precedence over Detach in that case.

Given that `select+case` statements are pseudo-random, is there a Go way of doing that? I could check to see (in the l.detached case) if l.close is also closed. Would that be the right way of doing it?

Reply:
Right i think that's a minor issue.
Your implementation would work.

I do think that a channel would allow the errors to be surfaced as they come, and so the consumer can handle them right away, and not get one or the other.